enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -659,6 +662,33 @@ func (l *FileCacheLoader) CacheSnapshot(ctx context.Context, key *fcpb.SnapshotK
 	return l.cacheActionResult(ctx, key, ar, opts)
 }
 
+// For merge queue branches, we should save the snapshot to the default branch (like `main`),
+// to support our fallback branch behavior.
+//
+// By default, if there is not a snapshot for the current branch,
+// we will fallback to a snapshot for the default branch.
+// If someone is using merge queues, they're unlikely to run CI on the default
+// branch itself, so there will never be a fallback branch hit. Instead, we should
+// fallback to the latest merge queue snapshot.
```

Comment:
> If someone is using merge queues, they're unlikely to run CI on the default branch itself

This is a bit surprising - is this what we're seeing in practice?

Reply:
Talked offline but TLDR:

I figured that if a customer is running CI on a merge branch, which will functionally become their main branch immediately after CI passes, there is no need to then run CI again on the main branch as it will be the same